tally_ho/apps/tally/views/reports/administrative_areas_reports.py

```diff
@@ -100,158 +323,483 @@ class RegionsReportsView(LoginRequiredMixin,
+                'center-and-stations-list',
+                tally_id=tally_id,
+                region_id=region_id)
+
+        if report_type_ == 'votes-per-candidate-report':
+            self.request.session['result_ids'] =\
+                list(progressive_report
+                     .filter(
+                         result_form__center__office__region__id=region_id)
+                     .values_list(
+                         'id', flat=True))
+            self.request.session['ballot_report'] = False
+
+            return redirect(
+                'candidate-list-by-votes',
+                tally_id=tally_id,
+                region_id=region_id)
 
         return self.render_to_response(
             self.get_context_data(
                 tally_id=tally_id,
-                report_name=_(u"Region"),
-                turn_out_report_download_url="regions-turnout-csv",
-                summary_report_download_url="regions-summary-csv",
+                administrative_area_name=None,
+                region_name=None,
+                constituency_name=None,
+                report_name=_(u'Region'),
+                administrative_area_child_report_name=_(
+                    u'Region Constituencies'),
+                turn_out_report_download_url='regions-turnout-csv',
+                summary_report_download_url='regions-summary-csv',
                 turnout_report=turnout_report,
-                summary_report=summary_report))
+                summary_report=summary_report,
+                progressive_report=progressive_report,
+                admin_ares_with_forms_in_audit=regions_with_forms_in_audit,
+                centers_stations_under_invg=centers_stations_under_invg,
+                centers_stations_ex_after_invg=centers_stations_ex_after_invg,
+                regions_report_url='regions-discrepancy-report',
+                child_turnout_report_url='constituency-turnout-report',
+                child_summary_report_url='constituency-summary-report',
+                child_discrepancy_report_url=str(
+                    'constituency-discrepancy-report'
+                ),
+                child_progressive_report_url=str(
+                    'constituency-progressive-report'
+                ),
+                admin_area_votes_per_candidate_report_url=str(
+                    'region-votes-per-candidate'
+                )))
 
 
 class ConstituencyReportsView(LoginRequiredMixin,
                               mixins.GroupRequiredMixin,
                               TemplateView):
     group_required = groups.TALLY_MANAGER
-    template_name = 'reports/administrative_areas_reports.html'
 
     def get(self, request, *args, **kwargs):
         tally_id = kwargs['tally_id']
-        export_type_ = kwargs.get('export_type')
+        region_id = kwargs['region_id']
+        report_type = kwargs.get('report_type', None)
+        constituency_id = kwargs.get('constituency_id', None)
+
+        region_name =\
```

Comment:
think this is clearer with a conjunctive, and to be consistent I'm now thinking we should change the above

```python
region_name = region_id and Region.objects.get(...
```

Reply:
Done